main.go

```diff
@@ -16,96 +15,168 @@ import (
+		Name:    "color",
+		Value:   "auto",
+		Usage:   "display logs in color. Possible values: auto, never.",
+		EnvVars: []string{"MARK_COLOR"},
+	}),
+	altsrc.NewBoolFlag(&cli.BoolFlag{
+		Name:    "debug",
+		Value:   false,
+		Usage:   "enable debug logs.",
+		EnvVars: []string{"MARK_DEBUG"},
+	}),
+	altsrc.NewBoolFlag(&cli.BoolFlag{
+		Name:    "trace",
+		Value:   false,
+		Usage:   "enable trace logs.",
+		EnvVars: []string{"MARK_TRACE"},
+	}),
+	altsrc.NewStringFlag(&cli.StringFlag{
+		Name:    "username",
+		Aliases: []string{"u"},
+		Value:   "",
+		Usage:   "use specified username for updating Confluence page.",
+		EnvVars: []string{"MARK_USERNAME"},
+	}),
+	altsrc.NewStringFlag(&cli.StringFlag{
+		Name:    "password",
+		Aliases: []string{"p"},
+		Value:   "",
+		Usage:   "use specified token for updating Confluence page. Specify - as password to read password from stdin, or your Personal access token. Username is not mandatory if personal access token is provided. For more info please see: https://developer.atlassian.com/server/confluence/confluence-server-rest-api/#authentication.",
+		EnvVars: []string{"MARK_PASSWORD"},
+	}),
+	altsrc.NewStringFlag(&cli.StringFlag{
+		Name:    "target-url",
+		Aliases: []string{"l"},
+		Value:   "",
+		Usage:   "edit specified Confluence page. If -l is not specified, file should contain metadata (see above).",
+		EnvVars: []string{"MARK_TARGET_URL"},
+	}),
+	altsrc.NewStringFlag(&cli.StringFlag{
+		Name:    "base-url",
+		Aliases: []string{"b"},
+		Value:   "",
+		Usage:   "base URL for Confluence. Alternative option for base_url config field.",
+		EnvVars: []string{"MARK_BASE_URL"},
+	}),
+	&cli.StringFlag{
+		Name:    "config",
+		Aliases: []string{"c"},
+		Value:   filepath.Join(os.Getenv("HOME"), ".config/mark"),
+		Usage:   "use the specified configuration file.",
+		EnvVars: []string{"MARK_CONFIG"},
+	},
+	altsrc.NewBoolFlag(&cli.BoolFlag{
+		Name:    "ci",
+		Value:   false,
+		Usage:   "run on CI mode. It won't fail if files are not found.",
+		EnvVars: []string{"MARK_CI"},
+	}),
+	altsrc.NewStringFlag(&cli.StringFlag{
+		Name:    "space",
+		Value:   "",
+		Usage:   "use specified space key. If the space key is not specified, it must be set in the page metadata.",
+		EnvVars: []string{"MARK_SPACE"},
+	}),
+}
+
 func main() {
-	cmd, err := docopt.ParseArgs(os.ExpandEnv(usage), nil, version)
-	if err != nil {
-		panic(err)
+	app := &cli.App{
+		Name:        "mark",
+		Usage:       usage,
+		Description: description,
+		Version:     version,
+		Flags:       flags,
+		Before: altsrc.InitInputSourceWithContext(flags,
+			func(context *cli.Context) (altsrc.InputSourceContext, error) {
+				if context.IsSet("config") {
+					filePath := context.String("config")
+					return altsrc.NewTomlSourceFromFile(filePath)
+				} else {
+					// Fall back to default if config is unset
+					return altsrc.NewTomlSourceFromFile(filepath.Join(os.Getenv("HOME"), ".config/mark"))
```

Comment:
I believe here we can try to read data from TOML, and somehow convert under_score keys to arg-like-params. This would lead to no changes in users' workflow.

Reply:
@kovetskiy I've added additional aliases to the flags, which should make it backward compatible.